frontend/app/blueprints/public/static/js/basket.js

```diff
@@ -0,0 +1,136 @@
+// functions relating to the sample basket and navbar
+async function openGrapeTree({newick, sampleIds, clusterMethod}) {
+    // Open grape tree
+    const form = document.getElementById("open-tree-form")
+    // add metadata and newick file to form
+    const newickInput = form.querySelector("#newick-content")
+    newickInput.setAttribute("value", newick)
+    const metadataInput = form.querySelector("#metadata-content")
+    const typingDataInput = form.querySelector("#typing-data-content")
+    typingDataInput.setAttribute("value", clusterMethod)
+    metadataInput.setAttribute("value", JSON.stringify({sample_id: sampleIds}))
+    const submitBnt = form.querySelector("input[type=submit]")
+    submitBnt.click()
+}
+
+// cluter all samples in basket
+async function clusterSamplesInBasket(element) {
+    const typingMethod = element.getAttribute("data-bi-typing-method") 
+    // base dropdown element
+    const baseElement = document.querySelector("#basket-cluster-samples")
+    const btn = baseElement.querySelector(".btn")
+    const clusterApiRoute = baseElement.getAttribute("data-bi-cluster-route") 
+    const sampleIds = JSON.parse(baseElement.querySelector("input[name=sample-ids]").value)
+    // construct body to pass
+    let body
+    switch (typingMethod) {
+        case "cgmlst":
+            body = {
+                sample_ids: sampleIds,
+                typing_method: typingMethod,
+                cluster_method: "MSTreeV2",
+            }
+            break
+        case "mlst":
+            body = {
+                sample_ids: sampleIds,
+                typing_method: typingMethod,
+                cluster_method: "MSTreeV2",
+             }
+             break
+        case "minhash":
+            body = {
+                sample_ids: sampleIds,
+                typing_method: typingMethod,
+                cluster_method: "single",
+            }
+            break
+    }
+    // submit job to API
+    showSpinner(btn)
+    try {
+        const response = await fetch(clusterApiRoute, {
+            method: "POST",
+            headers: { 
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(body)
+        })
+        if (!response.ok) {
+            throw new Error("Network response was not OK!")
+        }
+        const jobInfo = await response.json()
+        // inform user that clustering has started
+        throwSmallToast(`Clustering samples: ${sampleIds.length}`, "info")
+
+        // start polling for updates
+        let result = await poll(
+            async () => fetchJobStatus(jobInfo.id),  // GET job status
+            validateJobStatus,  // validator
+            resultParser,       // parse results
+            3000                // interval time
```

Comment:
I saw this in the previous JS as well. Could it be a global constant? If you want to use the same through the app, and it might change later